Week 17/PersonControllerTest/UnitTest1.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Week17.Domain;
+using Week17.Data;
+using Week17.Services;
+using Week17.Helpers;
+using Microsoft.Extensions.Options;
+using Week17.Controllers;
+using Microsoft.AspNetCore.Mvc;
+using Xunit;
+
+namespace Week17_tests
+{ 
+    public class Week17ControllerTest
+    {
+        public readonly PersonController _controller;
+        public IUserService _service;
+        public readonly PersonContext _context;
+        readonly AppSettings _appSettings;
+
+
+
+        public Week17ControllerTest()
+        {
+            _service = new UserServiceFake();
+            _context = new PersonContext();
+            _controller = new PersonController(_context, _service, (IOptions<AppSettings>)_appSettings);//მოკლედ ეს ვერ
```

Comment:
კონტროლერის კონსტრუქრუქტორი იღებს პარამეტრად აპპსეთინგსაც ან მაგაზე დეფენდენსი უნდა მოუშალო 
ან ტესტ ფაილში წააკითხო აპპსეთინგის ფაილიდან და ეგ გადასცე 
რადგანაც წესით ყველაფერი სერვისებში უნდა იყო გადატანილი კონტროლერში აპპსეთინგზე დეფენდენსი ზედმეტია და  მაგის გაქრობა უკეთესი იქნება

Reply:
არაა პრობლემა მთავარია იდეა გესმის 

https://dev.to/patzistar/c-unit-tests-with-mocks-8i7

mock  შეგიძლია გამოიყენო ფინალურზე და უფრო მარტივად გამოგივა